operator/chapter_os-migrate-vmware-guide.adoc

```diff
@@ -24,6 +24,168 @@ a very number of virtual machines as entry point, or can migrate sensitive virtu
 a near zero down time with the change block tracking VMWare option (CBT) and so perform the virtual
 machine migration in two steps. The migration can also be done without conversion host.
 
+== Pre-requisites and Validation Checks
+
+Before migrating VMware workloads to OpenStack, ensure the following pre-requisites are met.
+These checks are based on issues encountered in real-world migration engagements.
+
+=== Disk Consolidation
+
+Ensure that all virtual machine disks are consolidated before migration.
+
+Virtual machines with unconsolidated disks may fail during migration or result in data inconsistencies.
+Disk consolidation merges redundant delta disks created by snapshots back into the base disk.
+
+To check if disks need consolidation:
+
+* In vSphere Client, check the VM summary tab for "Consolidation needed" warnings
+* Via PowerCLI: `Get-VM | Where-Object {$_.ExtensionData.Runtime.ConsolidationNeeded}`
+
+To consolidate disks:
+
+* In vSphere Client: Right-click the VM → Snapshots → Consolidate
+* Ensure the operation completes successfully before proceeding with migration
+
+=== Snapshot Hierarchy Depth
+
+Verify that the snapshot hierarchy is not too deep (recommended maximum: 2-3 levels).
+
+Deep snapshot hierarchies can cause:
+
+* Performance degradation during migration
+* Increased risk of migration failures
+* Longer consolidation times
+
+To check snapshot hierarchy:
+
+* In vSphere Client: Right-click the VM → Snapshots → Manage Snapshots
+* Review the snapshot tree depth
+* If the hierarchy is too deep, consider consolidating or removing unnecessary snapshots before migration
+
+=== VMware User Access Control Lists (ACLs)
+
+Ensure the VMware user account has the correct permissions to perform migration operations.
+
+Required privileges:
+
+* **Virtual machine permissions:**
+** Virtual machine.Interaction.Power Off
+** Virtual machine.Interaction.Power On
+** Virtual machine.Provisioning.Allow disk access
+** Virtual machine.Provisioning.Allow read-only disk access
+** Virtual machine.State.Create snapshot
+** Virtual machine.State.Remove snapshot
+
+* **Datastore permissions:**
+** Datastore.Browse datastore
+** Datastore.Low level file operations
+
+* **General permissions:**
+** System.Anonymous
+** System.Read
+** System.View
+
+For CBT-based migrations, additional privileges may be required:
+
+** Virtual machine.Configuration.Change tracking
```

Comment:
I'm not sure those privileges really exists in vphere 7 and 8. From which version does it come from ?